packages/boot/test/bootstrapTests/test-vat-orchestration.ts

```diff
@@ -0,0 +1,102 @@
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import type { ExecutionContext, TestFn } from 'ava';
+import { createRequire } from 'module';
+import { BridgeId } from '@agoric/internal';
+import type { start as stakeAtomStart } from '@agoric/orchestration/src/contracts/stakeAtom.contract.js';
+import { makeWalletFactoryContext } from './walletFactory.ts';
+import { makeBridge } from './ibcBridgeMock.js';
+
+const { assign, entries } = Object;
+
+const nodeRequire = createRequire(import.meta.url);
+const assets = {
+  ibcServerMock: nodeRequire.resolve('./ibcServerMock.js'),
+  ibcClientMock: nodeRequire.resolve('./ibcClientMock.js'),
+};
+
+const makeTestContext = async (t: ExecutionContext) => {
+  const { bridgeHandler } = makeBridge(t);
+  return makeWalletFactoryContext(t, {
+    [BridgeId.DIBC]: obj => bridgeHandler.toBridge(obj),
+  });
+};
+
+type DefaultTestContext = Awaited<ReturnType<typeof makeTestContext>>;
+type StakeAtomPublicFacet = Awaited<
+  ReturnType<typeof stakeAtomStart>
+>['publicFacet'];
+
+const test: TestFn<DefaultTestContext> = anyTest;
+
+test.before(async t => (t.context = await makeTestContext(t)));
+test.after.always(t => t.context.shutdown?.());
+
+test('provideAccount returns an ICA connection', async t => {
+  const {
+    buildProposal,
+    bundleCache,
+    evalProposal,
+    runUtils: { EV },
+    installations,
+  } = t.context;
+  /** ensure network, ibc, and orchestration are available */
+  await evalProposal(
+    buildProposal('@agoric/builders/scripts/vats/init-network.js'),
+  );
+  await evalProposal(
+    buildProposal('@agoric/builders/scripts/vats/init-orchestration.js'),
+  );
+  const vatStore = await EV.vat('bootstrap').consumeItem('vatStore');
+  t.true(await EV(vatStore).has('ibc'), 'ibc');
+  t.true(await EV(vatStore).has('network'), 'network');
+  t.true(await EV(vatStore).has('orchestration'), 'orchestration');
+
+  /** ensure mock ibc services are available */
+  const zoe: ZoeService = await EV.vat('bootstrap').consumeItem('zoe');
+  for (const [name, path] of entries(assets)) {
+    const bundle = await bundleCache.load(path, name);
+    const installationRef = await EV(zoe).install(bundle);
```

Comment:
we try to avoid `E(zoe).install(bundle)` in anything but toys. This seems to be a pretty high-fidelity test, so I suggest using `controller.validateAndInstallBundle(bundle)` and `E(zoe).installBundleID(bundleID, label)`.

oops... I see we didn't do this consistently in `test-net-ibc-upgrade.ts`. oh well.

Reply:
Thank you, TIL

Reply:
cf
 - https://github.com/Agoric/agoric-sdk/issues/8083
 - https://github.com/Agoric/agoric-sdk/issues/4565